src/awkward/contents/unionarray.py

```diff
@@ -1574,13 +1574,25 @@ def _remove_structure(
         self, backend: Backend, options: RemoveStructureOptions
     ) -> list[Content]:
         out = []
-        for i in range(len(self._contents)):
-            index = self._index[self._tags.data == i]
-            out.extend(
-                self._contents[i]
-                ._carry(index, False)
+
+        tags = numpy.atleast_1d(self._tags.raw(numpy))
+        index = numpy.atleast_1d(self._index.raw(numpy))
+
+        for i in range(len(tags)):
+            tag = tags[i]
+            idx_value = index[i]
+
+            content = (
+                self._contents[tag]
```

Comment:
Hm, I don't think this can ever work with typetracers, because `tag` will be a `unknown_scalar`.